src/convenience_api.rs

```diff
@@ -0,0 +1,275 @@
+                            stream_type: stype,
+                            index: i,
+                            handler: Rc::clone(&self.handle),
+                        });
+                    };
+                    self.video_streams = Some(streams);
+                },
+                MediaInfoStream::Audio => {
+                    let mut streams = Vec::new();
+                    for i in 0..self.handle.borrow_mut().count_get(stype) {
+                        streams.push(AudioStream {
+                            stream_type: stype,
+                            index: i,
+                            handler: Rc::clone(&self.handle),
+                        });
+                    };
+                    self.audio_streams = Some(streams);
+                },
+                MediaInfoStream::Text => {
+                    let mut streams = Vec::new();
+                    for i in 0..self.handle.borrow_mut().count_get(stype) {
+                        streams.push(TextStream {
+                            stream_type: stype,
+                            index: i,
+                            handler: Rc::clone(&self.handle),
+                        });
+                    };
+                    self.text_streams = Some(streams);
+                },
+                MediaInfoStream::Other => {
+                    let mut streams = Vec::new();
+                    for i in 0..self.handle.borrow_mut().count_get(stype) {
+                        streams.push(OtherStream {
+                            stream_type: stype,
+                            index: i,
+                            handler: Rc::clone(&self.handle),
+                        });
+                    };
+                    self.other_streams = Some(streams);
+                },
+                MediaInfoStream::Image => {
+                    let mut streams = Vec::new();
+                    for i in 0..self.handle.borrow_mut().count_get(stype) {
+                        streams.push(ImageStream {
+                            stream_type: stype,
+                            index: i,
+                            handler: Rc::clone(&self.handle),
+                        });
+                    };
+                    self.image_streams = Some(streams);
+                },
+                MediaInfoStream::Menu => {
+                    let mut streams = Vec::new();
+                    for i in 0..self.handle.borrow_mut().count_get(stype) {
+                        streams.push(MenuStream {
+                            stream_type: stype,
+                            index: i,
+                            handler: Rc::clone(&self.handle),
+                        });
+                    };
+                    self.menu_streams = Some(streams);
+                },
+                _ => continue,
+            }
+        }
+    }
+
+    pub fn video_streams(&self) -> Option<&Vec<VideoStream>> {
+        match self.video_streams {
```

Comment:
I think that could be `self.video_streams.as_ref()`

Reply:
Sorry, I meant the whole message body 😄 https://doc.rust-lang.org/std/option/enum.Option.html#method.as_ref